app/controllers/proposal_controller.rb

```diff
@@ -46,31 +46,28 @@ def create
                            event_role: 'submitter')
     @event.event_users.new(user: current_user,
                            event_role: 'speaker')
+    if @event.save
+      ahoy.track 'Event submission', title: 'New submission'
 
-    unless @event.save
+      flash[:notice] = 'Proposal was successfully submitted.'
+      redirect_to conference_program_proposal_index_path(@conference.short_title)
+    else
       flash[:error] = "Could not submit proposal: #{@event.errors.full_messages.join(', ')}"
       render action: 'new'
-      return
     end
-
-    ahoy.track 'Event submission', title: 'New submission'
-
-    flash[:notice] = 'Proposal was successfully submitted.'
-    redirect_to conference_program_proposal_index_path(@conference.short_title)
   end
 
   def update
     authorize! :update, @event
     @url = conference_program_proposal_path(@conference.short_title, params[:id])
 
-    if !@event.update(event_params)
+    if @event.update(event_params)
+      redirect_to(conference_program_proposal_index_path(conference_id: @conference.short_title),
+                  notice: 'Proposal was successfully updated.')
+    else
       flash[:error] = "Could not update proposal: #{@event.errors.full_messages.join(', ')}"
-      render action: 'new'
-      return
+      render action: 'edit'
     end
-
-    redirect_to(conference_program_proposal_index_path(conference_id: @conference.short_title),
-                notice: 'Proposal was successfully updated.')
   end
 
   def destroy
```

Comment:
To save the event in `destroy` method [we are doing](https://github.com/openSUSE/osem/pull/867/files#diff-8c13e9cfe1f5edf7d4923e633a08f790R84):
```
@event.save(validate: false)
```

When for reset (`withdraw`, `restart` etc), we are doing:
```
if @event.save
   # do something
else
  # do something else
end
```

I think we should be consistent. Should I change `destroy` save to `if else` block too?